### PR TITLE
remove NNlib

### PR DESCRIPTION
**Pull request by CarloLucibello**
remove dependence on NNlib.jl as discussed in different places. 
Most stuff removed will end up in https://github.com/FluxML/NNlib.jl/pull/286

@denizyuret should the `batchnorm`  function stay or go?

### CODE REVIEW

**Comment by denizyuret** [comment]
> @denizyuret should the `batchnorm` function stay or go?

cuDNN introduced a new interface which I wrapped in CUDA/lib/cudnn/normalization.jl. They still support the old interface though, so feel free to move batchnorm.jl to whereever needed.

**Comment by CarloLucibello** [comment]
this should be ready. @denizyuret could you take a look and see I didn't remove too much stuff?

File: test/cudnn/nnlib.jl [hunk 1]
```diff
@@ -1,139 +0,0 @@
...
-        @test testf(logsoftmax, x)
-        @test testf(∇logsoftmax, dy, x) # add y when NNlib implements it
-    end
-end
-
-@testset "Activations and Other Ops" begin
-    @test testf(CUDNN.cudnnAddTensor, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
-    @test testf(CUDNN.cudnnActivationForward!, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
-    # @denizyuret: no high level api for backward functions, see CUDA/lib/cudnn/README.md
-    # @test testf(CUDNN.cudnnActivationBackward, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
```
Comment (CarloLucibello) [inline test/cudnn/nnlib.jl:101]: @denizyuret should these stay in this repo?

**Comment by denizyuret** [comment]
> this should be ready. @denizyuret could you take a look and see I didn't remove too much stuff?

Looks good to me. I assume you are moving these to whereever they are needed? (NNlibCUDA, Flux etc)

File: test/cudnn/nnlib.jl (hunk 1 above)
Comment (denizyuret) [reply test/cudnn/nnlib.jl:101]: No need, these are tested in CUDA/test/cudnn/inplace.jl and activation.jl.

**Comment by CarloLucibello** [comment]
> I assume you are moving these to whereever they are needed? (NNlibCUDA, Flux etc)

yes.

@maleadt this should be ready (assuming the buildkite's julia 1.6 debug fail is unrelated)

**Comment by maleadt** [comment]
Thanks! That failure is unrelated, and should be addressed by https://github.com/JuliaGPU/CUDA.jl/pull/756. Let's merge this after that PR.

**Pull request merged**